lib/dialects/mysql/query.js

```diff
@@ -48,9 +48,20 @@ Query.prototype.run = function(sql, parameters) {
 
         reject(self.formatError(err));
       } else {
-        resolve(self.formatResults(results));
+        resolve(results);
       }
     }).setMaxListeners(100);
+  })
+  // Log warnings if we've got them.
+  .then(function(results){
+    if (results && results.warningCount > 0) {
+      return self.logWarnings(results);
```

Comment:
You could call `query.run` recursively here, setting benchmark and logging to false